Sources/ApplicativeRouter/SomePartialIsos.swift

```diff
@@ -0,0 +1,140 @@
+import Either
+import Foundation
+import Optics
+import Prelude
+
+extension Optional {
+  public enum iso {
+    /// A partial isomorphism `(A) -> A?` 
+    public static var some: PartialIso<Wrapped, Wrapped?> {
+      return PartialIso<Wrapped, Wrapped?>(
+        apply: { .some($0) },
+        unapply: { $0 }
+      )
+    }
+  }
+}
+
+/// Lifts a partial isomorphism `(A) -> B` to one `(A) -> B?`.
+public func opt<A, B>(_ f: PartialIso<A, B>) -> PartialIso<A, B?> {
+  return f >>> Optional.iso.some
+}
+
+extension PartialIso where A == String, B == Int {
+  /// An isomorphism between strings and integers.
+  public static var int: PartialIso {
+    return .init(
+      apply: Int.init,
+      unapply: String.init
+    )
+  }
+}
+
+// TODO: possible to document this in the applicative of Syntax?
+
+extension PartialIso where A == String, B == Bool {
+  /// An isomorphism between strings and booleans.
+  public static var bool: PartialIso {
+    return .init(
+      apply: {
+        $0 == "true" || $0 == "1" ? true
```

Comment:
I guess you can just do `apply: { $0 == "1" ? true : $0 == "0" ? false : Bool($0.lowercased()) }`

Reply:
@ilyapuchka  oh yeah, I think that would be equivalent! forgot that `Bool` has a constructor taking a string...